grpclog/component.go

```diff
@@ -0,0 +1,233 @@
+	}
+	return "", false
+}
+
+// InfoDepth performs an info log of args at depth to the component, conditioned
+// on the component's log level.
+func (c *ComponentData) InfoDepth(depth int, args ...interface{}) {
+	if c.level > levelInfo {
+		return
+	}
+	args = append([]interface{}{"[" + string(c.name) + "]"}, args...)
+	grpclog.InfoDepth(depth, args...)
+}
+
+// WarningDepth performs a warning log of args at depth to the component,
+// conditioned on the component's log level.
+func (c *ComponentData) WarningDepth(depth int, args ...interface{}) {
+	if c.level > levelWarning {
+		return
+	}
+	args = append([]interface{}{"[" + string(c.name) + "]"}, args...)
+	grpclog.WarningDepth(depth, args...)
+}
+
+// ErrorDepth performs an error log of args at depth to the component.
+func (c *ComponentData) ErrorDepth(depth int, args ...interface{}) {
+	args = append([]interface{}{"[" + string(c.name) + "]"}, args...)
+	grpclog.ErrorDepth(depth, args...)
+}
+
+// FatalDepth performs a fatal log of args at depth to the component and then
+// exits the application in accordance with the logger's fatal behavior.
+func (c *ComponentData) FatalDepth(depth int, args ...interface{}) {
+	args = append([]interface{}{"[" + string(c.name) + "]"}, args...)
+	grpclog.FatalDepth(depth, args...)
+}
+
+// Info performs an InfoDepth log at depth 0.
+func (c *ComponentData) Info(args ...interface{}) {
+	c.InfoDepth(0, args...)
+}
+
+// Warning performs a WarningDepth log at depth 0.
+func (c *ComponentData) Warning(args ...interface{}) {
+	c.WarningDepth(0, args...)
+}
+
+// Error performs an ErrorDepth log at depth 0.
+func (c *ComponentData) Error(args ...interface{}) {
+	c.ErrorDepth(0, args...)
+}
+
+// Fatal performs a FatalDepth log at depth 0.
+func (c *ComponentData) Fatal(args ...interface{}) {
+	c.FatalDepth(0, args...)
+}
+
+// Infof formats the string and performs and InfoDepth log at depth 0.
+func (c *ComponentData) Infof(format string, args ...interface{}) {
+	c.InfoDepth(0, fmt.Sprintf(format, args...))
+}
+
+// Warningf formats the string and performs and WarningDepth log at depth 0.
+func (c *ComponentData) Warningf(format string, args ...interface{}) {
+	c.WarningDepth(0, fmt.Sprintf(format, args...))
+}
+
+// Errorf formats the string and performs and ErrorDepth log at depth 0.
+func (c *ComponentData) Errorf(format string, args ...interface{}) {
+	c.ErrorDepth(0, fmt.Sprintf(format, args...))
+}
+
+// Fatalf formats the string and performs and FatalDepth log at depth 0.
+func (c *ComponentData) Fatalf(format string, args ...interface{}) {
+	c.FatalDepth(0, fmt.Sprintf(format, args...))
+}
+
+// V reports whether thbe verbosity level of the component is at least l.
+func (c *ComponentData) V(l int) bool {
+	return c.verbosity >= l
+}
+
+// Component creates a new component and returns it for logging. If a component
+// with the name already exists, nothing will be created and it will be
+// returned.
+func Component(componentName string) *ComponentData {
```

Comment:
If this can return a `LoggerV2`, then we don't need to export `ComponentData`; I think that would be ideal.  Channelz could type assert it to a depth logger, so I think it could keep working.

Reply:
Component now returns a DepthLoggerV2